contentcuration/contentcuration/frontend/channelList/views/Channel/__tests__/catalogList.spec.js

```diff
@@ -1,200 +1,261 @@
+    language: 'en',
+  },
+];
 
 const results = ['channel-1', 'channel-2'];
 
-function makeWrapper(computed = {}) {
-  const loadCatalog = jest.spyOn(CatalogList.methods, 'loadCatalog');
-  loadCatalog.mockImplementation(() => Promise.resolve());
-
-  const downloadCSV = jest.spyOn(CatalogList.methods, 'downloadCSV');
-  const downloadPDF = jest.spyOn(CatalogList.methods, 'downloadPDF');
+function makeWrapper(overrides = {}) {
+  const mockSearchCatalog = jest.fn(() => Promise.resolve());
 
-  const wrapper = mount(CatalogList, {
-    router,
-    store,
-    computed: {
-      page() {
-        return {
-          count: results.length,
-          results,
-        };
+  const store = new Store({
+    state: {
+      connection: {
+        online: overrides.offline ? false : true,
       },
-      ...computed,
     },
-    stubs: {
-      CatalogFilters: true,
+    getters: {
+      loggedIn: () => true,
+    },
+    actions: {
+      showSnackbar: jest.fn(),
+    },
+    modules: {
+      channel: {
+        namespaced: true,
+        state: {
+          channelsMap: {
+            'channel-1': mockChannels[0],
+            'channel-2': mockChannels[1],
+          },
+        },
+        getters: {
+          getChannels: state => ids => {
+            return ids.map(id => state.channelsMap[id]).filter(Boolean);
+          },
+          getChannel: state => id => state.channelsMap[id],
+        },
+        actions: {
+          getChannelListDetails: jest.fn(() => Promise.resolve(mockChannels)),
+        },
+      },
+      channelList: {
+        namespaced: true,
+        state: {
+          page: {
+            count: results.length,
+            results,
+            page_number: 1,
+            total_pages: 1,
+            next: null,
+            previous: null,
+            ...overrides.page,
+          },
+        },
+        actions: {
+          searchCatalog: mockSearchCatalog,
+        },
+      },
     },
   });
-  return [wrapper, { loadCatalog, downloadCSV, downloadPDF }];
-}
 
-describe('catalogFilterBar', () => {
-  let wrapper, mocks;
-
-  beforeEach(async () => {
-    [wrapper, mocks] = makeWrapper();
-    await wrapper.setData({ loading: false });
+  const router = new VueRouter({
+    routes: [
+      {
+        name: RouteNames.CATALOG_ITEMS,
+        path: '/catalog',
+      },
+      {
+        name: RouteNames.CATALOG_DETAILS,
+        path: '/catalog/:channelId',
+      },
+    ],
   });
 
-  it('should call loadCatalog on mount', () => {
-    [wrapper, mocks] = makeWrapper();
-    expect(mocks.loadCatalog).toHaveBeenCalled();
+  router.push({ name: RouteNames.CATALOG_ITEMS }).catch(() => {});
+
+  const renderResult = render(CatalogList, {
+    localVue,
+    store,
+    router,
+    stubs: {
+      CatalogFilters: true,
+    },
+    mocks: {
+      $tr: (key, params) => {
+        const translations = {
+          resultsText: params ? `${params.count} result${params.count !== 1 ? 's' : ''} found` : '',
```

Comment:
One example of how this adds some complexity and difficulty is this `resultsText` as well as the `channelSelectionCount`. You are mocking the translations based on the params, but in reality, you are also controlling the params in the test suite, and the only string that is rendered will be `'2 results found'` `'2 channels selected'` (which you have later on in the tests).